src/common/models/Cookie.class.js

```diff
@@ -766,6 +756,32 @@ class Cookie {
     return `${h}:${m}`;
   }
 
+  parseVoyageObj(type, voyageObj) {
+    let voyage = {};
+    if (!voyageObj[`${type}Date`]) {
```

Comment:
This is a bit clearer I think:

`    voyage[`${type}Date`] = voyageObj[`${type}Date`] || this.generateDate(
        voyageObj[`${type}Day`],
        voyageObj[`${type}Month`],
        voyageObj[`${type}Year`]
    );

    voyage[`${type}Time`] = voyageObj[`${type}Time`] || this.generateTime(
        voyageObj[`${type}Hour`],
        voyageObj[`${type}Minute`]
    );`

Reply:
This changes has been made.